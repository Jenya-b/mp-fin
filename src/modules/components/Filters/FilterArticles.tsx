import { IArticle } from 'services/types';
import { Filter, Input, Item, Label, List, Subtitle } from './Filter.styled';

interface FilterWeekProps {
  arrArticles: string[];
  allArticles: IArticle[];
  setArticleNameFilter: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const FilterArticles = ({
  arrArticles,
  allArticles,
  setArticleNameFilter,
}: FilterWeekProps) => (
  <Filter>
    <Subtitle>Артикулы:</Subtitle>
    <List>
      {allArticles.map(({ articleId, articleName }) => (
        <Item key={articleId}>
          <Label htmlFor={articleName}>
            <Input
              id={articleName}
              type="checkbox"
              checked={arrArticles.includes(articleName)}
              onChange={setArticleNameFilter}
            />{' '}
            {''}
            {articleName}
          </Label>
        </Item>
      ))}
    </List>
  </Filter>
);
