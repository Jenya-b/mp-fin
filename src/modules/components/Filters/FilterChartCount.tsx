import { FormControl, NativeSelect } from '@mui/material';
import { IArticle } from 'services/types';
import { Filter, Subtitle } from './Filter.styled';

interface FilterWeekProps {
  setCountChart: React.Dispatch<React.SetStateAction<number>>;
}

export const FilterChartCount = ({ setCountChart }: FilterWeekProps) => (
  <Filter>
    <Subtitle>Количество графиков:</Subtitle>
    <FormControl fullWidth>
      <NativeSelect
        // defaultValue={30}
        onChange={(e) => setCountChart(Number(e.target.value))}
      >
        <option value={1}>1</option>
        <option value={2}>2</option>
        <option value={4}>4</option>
      </NativeSelect>
    </FormControl>
  </Filter>
);
