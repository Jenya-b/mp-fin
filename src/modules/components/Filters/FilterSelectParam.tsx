import { FormControl, NativeSelect } from '@mui/material';
import { setLocalStorage } from 'utils';
import { Filter, Subtitle } from './Filter.styled';

interface FilterSelectParamProps {
  setParameter: React.Dispatch<React.SetStateAction<number>>;
  thisParameter: number;
  parameters: number[];
  title: string;
  nameLocalStorage?: string;
}

export const FilterSelectParam = ({
  setParameter,
  thisParameter,
  parameters,
  title,
  nameLocalStorage,
}: FilterSelectParamProps) => {
  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const { value } = event.target;
    setParameter(Number(value));
    if (nameLocalStorage) {
      setLocalStorage(nameLocalStorage, Number(value));
    }
  };

  return (
    <Filter>
      <Subtitle>{title}</Subtitle>
      <FormControl fullWidth>
        <NativeSelect defaultValue={thisParameter} onChange={handleChange}>
          {parameters.map((item) => (
            <option key={item} value={item}>
              {item}
            </option>
          ))}
        </NativeSelect>
      </FormControl>
    </Filter>
  );
};
