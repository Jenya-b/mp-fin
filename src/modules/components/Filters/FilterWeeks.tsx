import { IWeekWithParam } from 'services/types';
import { formatDateGeneral } from 'utils/formatDate';
import { Filter, Input, Item, Label, List, Subtitle } from './Filter.styled';

interface FilterWeekProps {
  arrWeeks: string[];
  allWeeks: IWeekWithParam[];
  setWeekIdFilter: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const FilterWeeks = ({ arrWeeks, allWeeks, setWeekIdFilter }: FilterWeekProps) => (
  <Filter>
    <Subtitle>Недели:</Subtitle>
    <List>
      {allWeeks.map(({ weekId, weekNumber, weekStart, weekEnd }) => (
        <Item key={weekId}>
          <Label htmlFor={weekId}>
            <Input
              id={weekId}
              type="checkbox"
              checked={arrWeeks.includes(weekId)}
              onChange={setWeekIdFilter}
            />{' '}
            {''}
            {`${weekNumber} (${formatDateGeneral(weekStart)} - ${formatDateGeneral(weekEnd)})`}
          </Label>
        </Item>
      ))}
    </List>
  </Filter>
);
