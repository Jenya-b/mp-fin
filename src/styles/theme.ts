import { ITheme } from 'interfaces/styled';
import { Colors } from 'constants/colors';

export const baseTheme: ITheme = {
  colors: {
    backgroundBase: Colors.MainBlue,
    backgroundPrimary: Colors.MainWhite,
    backgroundSecondary: Colors.MainGreenSecondary,
    backgroundTertiary: Colors.MainGreenPrimary,
    backgroundTelegramPrimary: Colors.AdditionalBlue,
    backgroundTelegramSecondary: Colors.AdditionalBlue,
    textBase: Colors.MainBlack,
    textPrimary: Colors.MainGreySecondary,
    textSecondary: Colors.MainGreyLight,
    textTertiary: Colors.MainGreyPrimary,
    textAttentionPrimary: Colors.MainBluePrimary,
    textAttentionSecondary: Colors.AdditionalPinkSecondary,
    borderPrimary: Colors.MainGreyLight,
    borderSecondary: Colors.MainGreenSecondary,
    borderTertiary: Colors.MainGreyTertiary,
  },
  sizes: {
    loginForm: {
      width: 390,
    },
    inputForm: {
      height: 55,
    },
    logoForm: {
      width: 590,
    },
    primaryBtn: {
      height: 55,
    },
    sidebar: {
      widthActive: 290,
      widthHidden: 90,
      logo: { height: 80 },
    },
    header: {
      height: 80,
    },
    footer: {
      height: 60,
    },
    notFound: {
      title: { fontSize: 56 },
    },
    errorFallback: {
      title: { fontSize: 56 },
    },
    settingInputFile: {
      width: 178,
      height: 178,
    },
    replenishmentBlock: {
      height: 239,
    },
    adminForm: {
      maxWidth: 900,
      width: 100,
    },
    telegramIcon: {
      width: 25,
      height: 25,
    },
  },
  indents: {
    loginForm: {
      paddingTop: 39,
      paddingRight: 45,
      paddingBottom: 39,
      paddingLeft: 45,
    },
    inputList: {
      marginTop: 25,
      rowGap: 15,
    },
    main: {
      paddingLeft: 290,
      paddingLeftHide: 90,
    },
    adminMenu: {
      paddingTop: 40,
      paddingRight: 25,
      paddingBottom: 40,
      paddingLeft: 30,
    },
    adminList: {
      rowGap: 10,
    },
    adminContainer: {
      marginTop: 30,
    },
    adminForm: {
      columnGap: 10,
    },
    adminSubtitle: {
      marginBottom: 10,
    },
    adminSearchBlock: {
      marginTop: 30,
    },
    queryTableRow: {
      paddingTop: 7,
      paddingBottom: 7,
    },
  },
  borders: {
    loginForm: { borderRadius: 10 },
    inputForm: { borderRadius: 10 },
    primaryBtn: { borderRadius: 10 },
    replenishment: { borderRadius: 10 },
  },
  transitions: {
    input: { time: 0.3 },
    button: { time: 0.3 },
  },
  // z-index
  order: {
    sidebar: 50,
    settingInputFile: 10,
    settingLogo: 1,
  },
  cursor: 'pointer',
  media: {
    extraLarge: '(max-width: 1279px)',
    large: '(max-width: 1023px)',
    medium: '(max-width: 767px)',
    small: '(max-width: 479px)',
  },
};
