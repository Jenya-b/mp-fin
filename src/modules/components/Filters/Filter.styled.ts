import styled from 'styled-components';
import { fontStylesCaption, fontStylesCaptionBig, fontStylesH2 } from 'styles/typography';

export const Filter = styled.div``;

export const Title = styled.h2`
  ${fontStylesH2}
`;

export const Subtitle = styled.h3`
  ${fontStylesCaptionBig}
`;

export const List = styled.ul`
  margin-top: 10px;
  padding-right: 20px;
  max-height: 300px;
  overflow: auto;
  display: flex;
  flex-direction: column;
  row-gap: 5px;
`;

export const Item = styled.li``;

export const Label = styled.label`
  ${fontStylesCaption}
`;

export const Input = styled.input``;
