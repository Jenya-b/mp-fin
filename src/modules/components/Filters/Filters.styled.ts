import styled from 'styled-components';
import { fontStylesCaption, fontStylesCaptionBig, fontStylesH2 } from 'styles/typography';

export const Filters = styled.div`
  grid-row: 1/3;
  grid-column: 1/2;
  display: flex;
  flex-direction: column;
  row-gap: 20px;
`;

export const Filter = styled.div`
  margin-top: 20px;
`;

export const Title = styled.h2`
  ${fontStylesH2}
`;

export const Subtitle = styled.h3`
  ${fontStylesCaptionBig}
`;

export const List = styled.ul`
  margin-top: 10px;
  padding-right: 20px;
  max-height: 300px;
  overflow: auto;
  display: flex;
  flex-direction: column;
  row-gap: 5px;
`;

export const Item = styled.li``;

export const Label = styled.label`
  ${fontStylesCaption}
`;

export const Input = styled.input``;
