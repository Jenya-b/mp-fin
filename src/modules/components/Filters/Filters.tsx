import { Dispatch, SetStateAction, ChangeEvent } from 'react';
import { IFiltersData } from 'services/types';
import { Filters, Title } from './Filters.styled';
import { FilterByArticles } from './FilterByArticles/FilterByArticles';
import { FilterBySelectParams } from './FilterBySelectParams/FilterBySelectParams';
import { FilterByWeeks } from './FilterByWeeks/FilterByWeeks';
import { getListOfYears } from 'utils';

interface FiltersBlockProps {
  setCountChart: Dispatch<SetStateAction<number>>;
  countChart: number;
  countChartParam: number[];
  weekIdFilter: string[];
  filtersData: IFiltersData | undefined;
  setWeekIdFilter: (event: ChangeEvent<HTMLInputElement>) => void;
  articleNameFilter: string[];
  setArticleNameFilter: (event: ChangeEvent<HTMLInputElement>) => void;
  analiticYear: number;
  setAnaliticYear: Dispatch<SetStateAction<number>>;
}

export const FiltersBlock = ({
  setCountChart,
  countChart,
  countChartParam,
  weekIdFilter,
  filtersData,
  setWeekIdFilter,
  articleNameFilter,
  setArticleNameFilter,
  analiticYear,
  setAnaliticYear,
}: FiltersBlockProps) => {
  return (
    <Filters>
      <Title>Фильтр</Title>
      <FilterBySelectParams
        setParameter={setCountChart}
        thisParameter={countChart}
        parameters={countChartParam}
        title="Количество графиков:"
        nameLocalStorage="countChart"
        isFullWidth={true}
      />
      <FilterBySelectParams
        setParameter={setAnaliticYear}
        thisParameter={analiticYear}
        parameters={getListOfYears()}
        title="Выберите год:"
        nameLocalStorage="analiticYear"
        isFullWidth={true}
      />
      <FilterByWeeks
        arrWeeks={weekIdFilter}
        allWeeks={filtersData ? filtersData.weeksList : []}
        setWeekIdFilter={setWeekIdFilter}
        analiticYear={analiticYear}
      />
      <FilterByArticles
        arrArticles={articleNameFilter}
        allArticles={filtersData ? filtersData.articles : []}
        setArticleNameFilter={setArticleNameFilter}
      />
    </Filters>
  );
};
