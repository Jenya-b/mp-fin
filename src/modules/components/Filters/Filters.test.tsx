import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { baseTheme } from 'styles/theme';
import { FiltersBlock } from './Filters';

const props = {
  setCountChart: () => {},
  setWeekIdFilter: () => {},
  setArticleNameFilter: () => {},
  countChart: 1,
  countChartParam: [1, 2],
  weekIdFilter: ['id1', 'id2'],
  articleNameFilter: ['art1', 'art2'],
  filtersData: {
    articles: [
      {
        articleId: 'string',
        articleName: 'string',
        costPrice: 1,
        itemCode: 'string',
      },
    ],
    weeksList: [
      {
        weekEnd: 'string',
        weekNumber: 1,
        weekStart: 'string',
        weekId: 'string',
      },
    ],
  },
};

describe('FiltersBlock component', () => {
  test('FiltersBlock render', () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <FiltersBlock {...props} />
      </ThemeProvider>
    );
    expect(screen.getByText('Фильтр')).toBeInTheDocument();
  });
});
