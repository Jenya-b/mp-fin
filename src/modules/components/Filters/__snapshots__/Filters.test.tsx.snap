// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersBlock component FiltersBlock snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe hMmqpl"
      >
        <h2
          class="sc-dkrFOg dvcPFN"
        >
          Фильтры
        </h2>
        <div
          class="sc-gswNZR eEdGWU"
        >
          <h3
            class="sc-hLBbgP hnJdxU"
          >
            Количество графиков:
          </h3>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-v4u5dn-MuiInputBase-root-MuiInput-root"
            >
              <select
                aria-invalid="false"
                class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiInput-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
              >
                <option
                  selected=""
                  value="1"
                >
                  1
                </option>
                <option
                  value="2"
                >
                  2
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="sc-gswNZR eEdGWU"
        >
          <h3
            class="sc-hLBbgP hnJdxU"
          >
            Выберите год:
          </h3>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-v4u5dn-MuiInputBase-root-MuiInput-root"
            >
              <select
                aria-invalid="false"
                class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiInput-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
              >
                <option
                  selected=""
                  value="2023"
                >
                  2023
                </option>
                <option
                  value="2022"
                >
                  2022
                </option>
                <option
                  value="2021"
                >
                  2021
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="sc-gswNZR eEdGWU"
        >
          <h3
            class="sc-hLBbgP hnJdxU"
          >
            Недели:
          </h3>
          <ul
            class="sc-eDvSVe fUtpdY"
          />
        </div>
        <div
          class="sc-gswNZR eEdGWU"
        >
          <h3
            class="sc-hLBbgP hnJdxU"
          >
            Артикулы:
          </h3>
          <ul
            class="sc-eDvSVe fUtpdY"
          >
            <li>
              <label
                class="sc-jSUZER ljnyRR"
                for="string"
              >
                <input
                  id="string"
                  type="checkbox"
                />
                 
                string
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe hMmqpl"
    >
      <h2
        class="sc-dkrFOg dvcPFN"
      >
        Фильтры
      </h2>
      <div
        class="sc-gswNZR eEdGWU"
      >
        <h3
          class="sc-hLBbgP hnJdxU"
        >
          Количество графиков:
        </h3>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <select
              aria-invalid="false"
              class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiInput-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
            >
              <option
                selected=""
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
            </select>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-gswNZR eEdGWU"
      >
        <h3
          class="sc-hLBbgP hnJdxU"
        >
          Выберите год:
        </h3>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <select
              aria-invalid="false"
              class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiInput-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
            >
              <option
                selected=""
                value="2023"
              >
                2023
              </option>
              <option
                value="2022"
              >
                2022
              </option>
              <option
                value="2021"
              >
                2021
              </option>
            </select>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-gswNZR eEdGWU"
      >
        <h3
          class="sc-hLBbgP hnJdxU"
        >
          Недели:
        </h3>
        <ul
          class="sc-eDvSVe fUtpdY"
        />
      </div>
      <div
        class="sc-gswNZR eEdGWU"
      >
        <h3
          class="sc-hLBbgP hnJdxU"
        >
          Артикулы:
        </h3>
        <ul
          class="sc-eDvSVe fUtpdY"
        >
          <li>
            <label
              class="sc-jSUZER ljnyRR"
              for="string"
            >
              <input
                id="string"
                type="checkbox"
              />
               
              string
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
